meson_options.txt

```diff
@@ -17,3 +17,10 @@ option(
   description: 'Directory for system-wide tmpfiles.d(5) files',
   type: 'string',
 )
+
+option(
+  'install_completions',
```

Comment:
Umm... do we need a new build flag for completions?

With this pull request, we now have several subtly different knobs for having shell completions.  First, there's the `install_completions` build flag, then we have the optional dependencies on `bash-completion` and `fish`, but we have some fallbacks in case they are missing.  Documenting those three layers is going to be needlessly more difficult in comparison to what [we had before](https://containertoolbx.org/install/):
```
The following dependencies enable various optional features:
  * bash-completion
```

Then we would have to test those layers.

Generally speaking, we shouldn't be adding build options and fallbacks without well-defined use-cases.

Reply:
My aim with providing fallbacks was to make sure we're always installing in the "correct" locations. But looking at what Podman [does](https://github.com/containers/podman/blob/main/Makefile#L76-L78), we could probably just drop the fallbacks and hard-code the paths reducing the number of layers.

I'd rather go with the approach of having an explicit switch for completions instead of having to require said shells.

Reply:
We can't be correct if we are guessing.  If a shell's completion machinery offers an API for the location, then we should be using only that.  eg., how can we know that the right location for fish is `/usr/share/fish/completions` or `/usr/share/fish/vendor_completions.d`?  Any random downstream distributor or a new release of the shell can switch the location and we would be in trouble.

In other words, I don't think Podman is doing the right thing.  :)

> I'd rather go with the approach of having an explicit switch for completions
> instead of having to require said shells.

There are three groups of people building Toolbx:
* Downstream distributors shipping `toolbox` binaries
* Continuous integration
* Developers hacking on the sources

For the first, it's acceptable to require the shells as a build dependency so that we can use the appropriate API to get the location for completions.  However, it doesn't mean that the shells need to be run-time dependencies of `toolbox`.  eg., I have fish and zsh completions installed for `flatpak` and `podman` on my Fedora 36 even though I don't have the `fish` and `zsh` RPMs.

The continuous integration case seems to be the same as the first.

Developers hacking on Toolbx, who aren't specifically working on the completions, don't need to worry about the completions at all.  They likely don't even want to, because their running shells won't pick up the newly installed completions from a random prefix.  Even more so if their host OS is Fedora Silverblue because they would be hacking inside a Toolbx container.  So, it's fine if the build skips over the completions because the necessary build dependencies were missing.

If a developer is indeed hacking on the completions, then it's reasonable to expect them to have the necessary shells installed.  Otherwise how would they test their changes?  :)

Reply:
I did some more digging.

> In other words, I don't think Podman is doing the right thing. :)

Another incorrect example is Flatpak.  :)

A good example to emulate would be systemd:
* https://github.com/systemd/systemd/blob/main/shell-completion/bash/meson.build
* https://github.com/systemd/systemd/blob/main/shell-completion/zsh/meson.build

We should default to using the shell's API to detect the location for installing the completions, but have a build flag to override that.  Just like we do for `tmpfiles_dir`.

This is particularly relevant for Z shell, because it doesn't have an API and the Internet says that Debian and Fedora use [different locations](https://unix.stackexchange.com/questions/607809/where-should-i-install-the-zsh-completion-file-as-a-package-maintainer).  A build flag for the location can also [optimize the build](https://src.fedoraproject.org/rpms/toolbox/c/9bebde5bb60f36e38324c1e230e457c495d2d864) when there's an alternate API for the location.

Reply:
I have tried this in https://github.com/containers/toolbox/pull/1123